CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/recipes/MCRecipeManager.java

```diff
@@ -301,132 +304,160 @@ public void removeRecipes(List<ResourceLocation> removingRecipes) {
             removingRecipes.forEach(recipe -> RegistryManager.ACTIVE.getRegistry(GameData.RECIPES).remove(recipe));
         }
     }
-    
-    public static class ActionReplaceAllOccurences extends ActionBaseRemoveRecipes {
-    
-        //I'm odd, in that I'm an ActionBaseRemoveRecipes, that also creates recipes.
+
+    public static class SubActionReplaceAllOccurences implements IAction {
+        private SubActionReplaceAllOccurences next;
         private final IIngredient toReplace;
         private final IIngredient replaceWith;
         private final IIngredient forOutput;
-        private List<MCRecipeBase> toChange;
-        private List<ResourceLocation> toRemove;
-        
-        public ActionReplaceAllOccurences(IIngredient toReplace, IIngredient replaceWith, IIngredient forOutput) {
+
+        public SubActionReplaceAllOccurences(IIngredient toReplace, IIngredient replaceWith, IIngredient forOutput) {
             this.toReplace = toReplace;
             this.replaceWith = replaceWith;
-            this.forOutput = forOutput == null ? IngredientAny.INSTANCE : forOutput;
+            this.forOutput = forOutput;
         }
-        
+
         @Override
         public void apply() {
-            toChange = getAllForIngredient(toReplace);
-            toRemove = toChange.stream()
-                    .map(f -> new ResourceLocation(f.getFullResourceName()))
-                    .collect(Collectors.toList());
-            removeRecipes(toRemove);
-            changeIngredients(toChange);
-        }
-        
-        @Override
-        public void removeRecipes(List<ResourceLocation> removingRecipes) {
-            List<ActionBaseAddRecipe> toUnAdd = new ArrayList<>();
-            removingRecipes.forEach(recipe -> {
-                RegistryManager.ACTIVE.getRegistry(GameData.RECIPES).remove(recipe);
-                recipesToAdd.stream()
-                        .filter(f -> f instanceof ActionDummyAddRecipe)
-                        .filter(f -> f.recipe.getRegistryName().equals(recipe))
-                        .forEach(toUnAdd::add);
-            });
-            toUnAdd.forEach(f -> {
-                recipesToAdd.remove(f);
-                usedRecipeNames.remove(f.getName());
-            });
+            if (ActionReplaceAllOccurences.INSTANCE.currentModifiedRecipe != null) {
+                if (forOutput == null || forOutput.contains(ActionReplaceAllOccurences.INSTANCE.currentModifiedRecipe.getOutput())) {
+                    if (ActionReplaceAllOccurences.INSTANCE.ingredients1D != null) {
+                        for (int i = 0; i < ActionReplaceAllOccurences.INSTANCE.ingredients1D.length; i++) {
```

Comment:
Please extract the body of this if-block into a separate method `applyShapeless()`